blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/processors/DiscoveryWriter.java

```diff
@@ -98,6 +98,13 @@ public void registerIOModules() {
     }
 
     public ListenableFuture<Boolean> processMetrics(final List<List<IMetric>> input) {
+        int remainingCapacityOfTheBlockingQueue = this.remainingCapacityOfTheQueue();
+        log.debug("Remaining capacity of DiscoveryWriter blocking queue: [" + remainingCapacityOfTheBlockingQueue + "]");
+
+        if(remainingCapacityOfTheBlockingQueue == 0){
```

Comment:
this will be true when the queue will be of length INT_MAX value. In fact, that's what is happening in Ingest node and it's bringing the whole node to grinding halt.